### ISSUE TITLE
iOS VO improvements for selection

### ISSUE THREAD

**Issue by snowystinger** [adobe/react-spectrum#2522]
# 🙋 Feature Request

iOS VO doesn't support grid cell or grid row announcements. Two areas we can improve:
1) Where should we put the accessibilityDescription for gridcell/gridrow? It does not announce in iOS VoiceOver - for long press. We need it to announce the “long press to blah blah”
2) How should we handle selection announcements in highlight selection mode so it doesn’t announce every time you move focus.

From testing session:
- we should probably have an announcement when entering into selection mode (DG)

## 🤔 Expected Behavior

Table's should say things like, 'long press to enter selection mode' and shouldn't too verbose to the point of being obnoxious when reporting selection changes if selection follows focus.

## 😯 Current Behavior

Nothing is announced for moving in and out of selection mode for TableView.

## 💁 Possible Solution

1. Maybe putting it on the whole table would work
2. Maybe somehow only announce selections that occur not after an arrow key navigation
